pkg/reconciler/ingress/config/gateway.go

```diff
@@ -98,24 +97,12 @@ func NewGatewayFromConfigMap(configMap *corev1.ConfigMap) (*Gateway, error) {
 	c := Gateway{Gateways: map[v1alpha1.IngressVisibility]*GatewayConfig{}}
```

Comment:
Ah now I see why you had to do the variable shadowing: Why are we copying the map anyway? Can we not "just" do `c.Gateways = entry`? Why does the GatewayConfig need to be a pointer?